### PR TITLE
Implement Ctrl-S shortcut for saving current file

### LINKED ISSUE #157: Ctrl-S should be overridden to save the current file instead
Ctrl-S in browser saves the webpage (HTML).
We want to instead save the currently edited file to saved in the user's computer. 

When pressed Ctrl-S, user should get a file browser dialog to save file, which would be the file he is editing. It should show appropriate filetype/extension (.c, .cpp, .java etc) depending on language.

### PR DESCRIPTION
**Pull request by kapillamba4**
Fixes #157 
Download and Upload works on both chrome and mozilla.
All 4 shortcuts work on both the browsers
Ctrl + X: Reset the code
Ctrl + S: Save to file
Ctrl + Z: Run the code
Ctrl + B: Reset the settings

### CODE REVIEW

**Comment by championswimmer** [comment]
This implementation seems better than #159 in totality, but I'd like to grant @architkshk some bounty points for keeping Mac in mind 

@kapillamba4 can you include support for Cmd-<key> in Mac, while keeping Ctrl-<key> as it is for other platforms ? 

I plan to split the bounty 
Archit: 200
Kapil 800

Please make claims accordingly.

**Comment by championswimmer** [comment]
Also please do not tinker with 

Cmd-Z/Ctrl-Z it works like undo. Let it be what it is
Cmd-X/Ctrl-X it works like cut. Let it be

You can use Ctrl-B, Ctrl-I, Ctrl-U those combinations have no use in code editors.

**Comment by kapillamba4** [comment]
Update the PR, new shortcuts are:
Cmd + B || Ctrl + B: Reset the code
Cmd + S || Ctrl + S: Save to file
Cmd + I || Ctrl + I: Run the code
Cmd + U || Ctrl + U: Reset the settings

**Comment by championswimmer** [comment]
Is it possible that the download file be generated with proper extension for source code type ?

**Comment by kapillamba4** [comment]
yup, i'll update the pull request. Shall i add a modal for user to enter or edit the File name when download button is clicked or Ctrl+S shortcut is invoked?

**Comment by kapillamba4** [comment]
yup, i'll update the pull request. Should i also add a modal to allow user to enter or edit the File name when download button is clicked or Ctrl+S shortcut is invoked?

File: src/store/index.js [hunk 1]
```diff
@@ -56,8 +56,18 @@ export default new Vuex.Store({
...
+        'C++': '.cpp',
+        'C#': '.cs',
+        'Java': '.java',
+        'Javascript': '.js',
+        'NodeJs': '.js',
+        'Ruby': '.rb'
+      }
       state.language = val
       state.languageMode = languageMode[state.language]
+      state.fileName = `download/${extension[state.language]}`
```
Comment (championswimmer) [inline src/store/index.js]: shouldn't it be '.' not '/' ?

Comment (kapillamba4) [reply src/store/index.js]: Updated the pull request, I'll also add a modal to prompt user to enter/confirm file name before downloading when I have time

**Pull request merged**